### PR TITLE
Update kubeflow guide

### PR DESCRIPTION
**Pull request by schustmi**
## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Describe changes
- Update the last chapter of the low level guide

### CODE REVIEW

**Review by alexej-zenml — APPROVED** [review]

File: docs/book/guides/low-level-api/chapter-7.md [hunk 1]
```diff
@@ -8,111 +8,122 @@ If you want to see the code for this chapter of the guide, head over to the [Git
...
-### Install and configure Airflow
+### Pre-requisites
 
-This part is optional, and it would depend on your pre-existing production setting. For example, for this guide, Airflow will be set up from scratch and set up to work locally, however you might want to use a managed Airflow instance like [Cloud Composer](https://cloud.google.com/composer) or [Astronomer](https://astronomer.io).
+In order to run this example, we have to install a few tools that allow ZenML to spin up a local Kubeflow Pipelines setup:
 
-For this guide, you'll want to make sure airflow is installed before continuing:
+* [K3D](https://k3d.io/v5.2.1/#installation) to spin up a local Kubernetes cluster
+* The Kubernetes command-line tool [Kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl) to deploy Kubeflow Pipelines
+* [Docker](https://docs.docker.com/get-docker/) to build docker images that run your pipeline in Kubernetes pods
```
Comment (alex-zenml) [inline docs/book/guides/low-level-api/chapter-7.md:20]:
```suggestion
* [Docker](https://docs.docker.com/get-docker/) to build Docker images that run your pipeline in Kubernetes pods
```

File: docs/book/guides/low-level-api/chapter-7.md [hunk 2]
```diff
@@ -8,111 +8,122 @@ If you want to see the code for this chapter of the guide, head over to the [Git
...
 
-```shell
-pip install apache_airflow==2.2.0
+{% hint style="warning" %}
+The local Kubeflow Pipelines deployment requires more than 2 GB of RAM, so if you're using Docker Desktop make sure to update the resource limits in the preferences
```
Comment (alex-zenml) [inline docs/book/guides/low-level-api/chapter-7.md:23]:
```suggestion
The local Kubeflow Pipelines deployment requires more than 2 GB of RAM, so if you're using Docker Desktop make sure to update the resource limits in the preferences.
```

File: docs/book/guides/low-level-api/chapter-7.md [hunk 3]
```diff
@@ -8,111 +8,122 @@ If you want to see the code for this chapter of the guide, head over to the [Git
...
 
-### Creating an Airflow Stack
+### Create a local Kubeflow Pipelines Stack
 
 A [Stack](../../core-concepts.md) is the configuration of the surrounding infrastructure where ZenML pipelines are run and managed. For now, a `Stack` consists of:
 
 * A metadata store: To store metadata like parameters and artifact URIs
 * An artifact store: To store interim data step output.
 * An orchestrator: A service that actually kicks off and runs each step of the pipeline.
+* An optional container registry: To store docker images that are created to run your pipeline.
```
Comment (alex-zenml) [inline docs/book/guides/low-level-api/chapter-7.md:41]:
```suggestion
* An optional container registry: To store Docker images that are created to run your pipeline.
```

File: docs/book/guides/low-level-api/chapter-7.md [hunk 4]
```diff
@@ -8,111 +8,122 @@ If you want to see the code for this chapter of the guide, head over to the [Git
...
+Stack `local_kubeflow_stack` successfully registered!
+Active stack: local_kubeflow_stack
 ```
 
-![Your stack with Airflow as orchestrator](../../.gitbook/assets/localstack-with-airflow-orchestrator.png)
+![Your stack with a Kubeflow Pipelines Orchestrator](../../.gitbook/assets/localstack-with-kubeflow-orchestrator.png)
 
 {% hint style="warning" %}
-In the real-world we would also switch to something like a MySQL-based metadata store and a Azure/GCP/S3-based artifact store. We have just skipped that part to keep everything in one machine to make it a bit easier to run this guide.
+In the real-world we would also switch to something like a MySQL-based metadata store and an Azure/GCP/S3-based artifact store. We have just skipped that part to keep everything in one machine to make it a bit easier to run this guide.
```
Comment (alex-zenml) [inline docs/book/guides/low-level-api/chapter-7.md:88]:
```suggestion
In the real world we would also switch to something like a MySQL-based metadata store and an Azure-, GCP-, or S3-based artifact store. We have just skipped that part to keep everything in one machine to make it a bit easier to run this guide.
```

File: docs/book/guides/low-level-api/chapter-7.md [hunk 5]
```diff
@@ -8,111 +8,122 @@ If you want to see the code for this chapter of the guide, head over to the [Git
...
-
-* APP\_DIR will depend on your os. See which path corresponds to your OS [here](https://click.palletsprojects.com/en/8.0.x/api/#click.get\_app\_dir).
-* STACK\_UUID will be the unique id of the airflow\_stack. There will be only one folder here so you can just navigate to the one that is present.
-{% endhint %}
 
-### Run
+### Run the pipeline
 
-The code from this chapter is the same as the last chapter. So run:
+There is one minor change we need to make to run the pipeline from the previous chapter: We need to specify the python package requirements that ZenML should install inside the docker image it creates for you.
```
Comment (alex-zenml) [inline docs/book/guides/low-level-api/chapter-7.md:101]:
```suggestion
There is one minor change we need to make to run the pipeline from the previous chapter: we need to specify the Python package requirements that ZenML should install inside the Docker image it creates for you. 
```

File: docs/book/guides/low-level-api/chapter-7.md [hunk 6]
```diff
@@ -8,111 +8,122 @@ If you want to see the code for this chapter of the guide, head over to the [Git
...
 ```
 
-Even through the pipeline script is the same, the output will be a lot different from last time. ZenML will detect that `airflow_stack` is the active stack, and do the following:
-
-* `chapter_7.py` will be copied to the Airflow `dag_dir` so Airflow can detect is as an [Airflow DAG definition file](https://airflow.apache.org/docs/apache-airflow/stable/tutorial.html#it-s-a-dag-definition-file).
-* The Airflow DAG will show up in the Airflow UI at [http://0.0.0.0:8080](http://0.0.0.0:8080). You will have to login with the username and password generated above.
-* The DAG name will be the same as the pipeline name, so in this case `mnist_pipeline`.
-* The DAG will be scheduled to run every minute.
-* The DAG will be un-paused so you'll probably see the first run as you click through.
+We can now run the pipeline by simply executing the python script:
```
Comment (alex-zenml) [inline docs/book/guides/low-level-api/chapter-7.md:109]:
```suggestion
We can now run the pipeline by simply executing the Python script:
```

**Review by alex-zenml — CHANGES_REQUESTED** [review]

**Review by htahir1 — APPROVED** [review]
Alex's changes aside, happy to merge!

**Pull request merged**